test/tools/jsontests/BlockChainTests.h

```diff
@@ -44,6 +44,52 @@ class bcGeneralTestsFixture : public StateTestFixtureBase<BCGeneralStateTestsSui
     bcGeneralTestsFixture() : StateTestFixtureBase({TestExecution::RequireOptionAll}) {}
 };
 
+template <class T>
```

Comment:
I think you can easy get rid of duplication here like this
```
BlockChainTestFixture
class BlockChainTestFixture
{
public:
    BlockChainTestFixture(std::set<TestExecution> const& _execFlags = {}, std::vector<std::string> const& _timeConsumingCases)
    {
        T suite;
        if (_execFlags.count(TestExecution::NotRefillable) &&
            (Options::get().fillchain || Options::get().filltests))
            BOOST_FAIL("Tests are sealed and not refillable!");

        string const casename = boost::unit_test::framework::current_test_case().p_name;
        boost::filesystem::path const suiteFillerPath = suite.getFullPathFiller(casename).parent_path();

        // skip time consuming tests, unless --all option is given
        if (!test::Options::get().all && contains(_timeConsumingCases, casename))
        {
            cnote << "Skipping " << casename << " because --all option is not specified.\n";
            test::TestOutputHelper::get().markTestFolderAsFinished(suiteFillerPath, casename);
            return;
        }

        suite.runAllTestsInFolder(casename);
        test::TestOutputHelper::get().markTestFolderAsFinished(suiteFillerPath, casename);
    }
};
```
```
class BlockChainValidBlocksTestFixture
  : public BlockChainTestFixture<BlockchainValidTestSuite>
{
public:
    BlockChainValidBlocksTestFixture()
      : BlockChainValidBlocksTestFixture({TestExecution::NotRefillable}, {"bcWalletTest"})
    {}
};

```

Reply:
Actually, if `stWallet` exists in only in either valid or invalid suites, you can use the exact same code with `stWallet` check for both suites (it won't skip `stWallet` if it doesn't exist there)

Reply:
I think it is ok to use just one fixture class in this case